### DISCUSSION TITLE
checking if a message mentions a specific user: problem with getMemberById and retrieveMemberById

### DISCUSSION THREAD

**Discussion by TheDartFrog** [discord-jda/JDA#2384] [Questions and Help] [answered]
hi! 
I wanted to make my bot react to my friend getting pinged in a specific channel by using this in an if statement: 
``event.getMessage().getMentionedMembers().contains(event.getGuild().getMemberById(friendUserId)) && event.getChannel().getId().equals("idOfSpecificChannel")``
this worked perfectly.. for a few hours, then getMemberById started returning null.. I read that getMemberById only checks cached users, so I figured the Member of my friend simply got uncached. I tried using retrieveMemberById following the example I found in [this post](https://github.com/DV8FromTheWorld/JDA/discussions/1725) yet I still am getting null if I try to getMember, and retrieveMember by itself appears to return the request to discord and not the Member itself.. following the post I linked earlier, I tried this: 
`event.getGuild().retrieveMemberById(fedexUserId, true).queue(member -> {
            System.out.println(event.getGuild().getMemberById(friendUserId));;
        }, new ErrorHandler().handle(ErrorResponse.UNKNOWN_MEMBER, error -> {
            System.out.println("bruhhhhhhhh");
        }));` 
where event is a MessageReceivedEvent, but getMemberById returns null still
        
I wanna add that getMentionedMembers() on a message where I ping my friend prints the Member of my friend just fine, perhaps it would be possible to somehow hardcode the member then instead of getting it by ID?
        
        
Earlier I was checking if  'getMentionedMembers().get(0).getIdLong()' matched the ID of my friend, but that of course only worked if my friend was the first one to be mentioned in the message, and I also didn't like the jankyness of that solution
        
I also know I could just check if the message contains <@IdLongOfMyFriend> but that feels kinda janky as well, and it also doesn't account for example for reply mentions 
        
        
Please excuse me if the problem is very obvious and it's just me not understand something, I'm still learning!

**Comment by TheDartFrog** [comment, marked as answer]
okay, I managed to solve it finally!
I had to add ".setMemberCachePolicy(ALL)" to the JDA builder in the Main function, I also am now making sure to retrieve the Member of my friend like this:

`@Override
    public void onReady(ReadyEvent readyEvent) {

        readyEvent.getJDA().getGuildById(1001558016119816192L).retrieveMemberById(friendUserId, true).queue(member -> {
            System.out.println(readyEvent.getJDA().getGuildById(1001558016119816192L).getMemberById(friendUserId));
        }, new ErrorHandler().handle(ErrorResponse.UNKNOWN_MEMBER, error -> {
            System.out.println("bruhhhhhhhh friend wher");
        }));

    }`

then later on getMember works just fine!

**Reply by MinnDevelopment** [reply]
You only have to check the id though:

```java
boolean isMentioned = message.getMentionedMembers().stream().anyMatch(m -> m.getIdLong() == friendUserId);
```
